slack-service/app.py

```diff
@@ -0,0 +1,182 @@
+    """Send a message to an agent and return the response"""
+    try:
+        response_stream = agent_manager._client.agents.turn.create(
+            agent_id=agent_id,
+            session_id=session_id,
+            stream=True,
+            messages=messages,
+        )
+
+        response = ""
+        for chunk in response_stream:
+            # check if the chunk contains an error attribute before accessing it
+            if hasattr(chunk, "error") and chunk.error:
+                error_message = chunk.error.get("message", "Unknown agent error")
+                print(f"Error from agent API: {error_message}")
+                return f"Error from agent: {error_message}"
+
+            # If no error, process the stream for the final message
+            if (
+                hasattr(chunk, "event")
+                and hasattr(chunk.event, "payload")
+                and chunk.event.payload.event_type == "turn_complete"
+            ):
+                if hasattr(chunk.event.payload.turn, "output_message"):
+                    content = chunk.event.payload.turn.output_message.content
+                    response += content
+
+        return response.strip()
+
+    except Exception as e:
+        print(f"Error in send_message_to_agent: {e}")
+        return f"Error: {str(e)}"
+
+
+def handle_user_message(user_id: str, text: str, channel_id: str):
+    """Handle incoming user message and route to appropriate agent"""
+
+    # Create or get user session
+    if user_id not in user_sessions:
+        routing_agent_id = available_agents.get("routing-agent")
+        if not routing_agent_id:
+            return "Error: Core routing agent not available.", 500
+
+        session = agent_manager._client.agents.session.create(
+            routing_agent_id, session_name=f"slack-session-{user_id}"
+        )
+        user_sessions[user_id] = {
+            "agent_id": routing_agent_id,
+            "session_id": session.session_id,
+        }
+        print(f"New session for user {user_id}")
+
+    current_session = user_sessions[user_id]
+    messages = [{"role": "user", "content": text}]
+
+    # Get initial response
+    agent_response = send_message_to_agent(
+        current_session["agent_id"], current_session["session_id"], messages
+    )
+
+    # Check if we need to route to a different agent
+    potential_agent_name = agent_response.strip()
+    if (
+        potential_agent_name in available_agents
+        and potential_agent_name != "routing-agent"
+        and current_session["agent_id"] == available_agents.get("routing-agent")
+    ):
+
+        print(f"Routing to agent: {potential_agent_name}")
+        new_agent_id = available_agents[potential_agent_name]
+
+        # Create new session with specific agent
+        new_session = agent_manager._client.agents.session.create(
+            new_agent_id,
+            session_name=f"slack-session-{user_id}-{potential_agent_name}",
```

Comment:
same comment here about the session always beeing the same even if we are in a different session with the user.